packages/plugin-ext/src/main/browser/view/tree-view-widget.tsx

```diff
@@ -389,17 +390,37 @@ export class TreeViewWidget extends TreeViewWelcomeWidget {
 
     handleClickEvent(node: TreeNode, event: React.MouseEvent<HTMLElement>): void {
         super.handleClickEvent(node, event);
-        this.tryExecuteCommand(node);
+        // If clicked on item (not collapsable icon) - execute command or toggle expansion if item has no command
+        if (!event.currentTarget.hasAttribute('data-node-id')) {
```

Comment:
Are you certain this check is necessary? As I read the code from TreeWidget that handles the toggle click
https://github.com/eclipse-theia/theia/blob/3bd43c98dc5c755de7ae23d3123a32e54096a4a1/packages/core/src/browser/tree/tree-widget.tsx#L536-L549

this method (`handleClickEvent`) should not be called on a click on the toggle once you set `expandOnlyOnExpansionToggleClick` to `true`. Did you find that it was being called anyway?

Reply:
Yeah, you right, it is unnecessary, thanks! I removed this check and tested again - it still works as expected.